platform/ios/src/MGLMapView.mm

```diff
@@ -3995,20 +4000,21 @@ - (void)selectAnnotation:(id <MGLAnnotation>)annotation animated:(BOOL)animated
             MGLAnnotationContext &annotationContext = _annotationContextsByAnnotationTag.at(annotationTag);
             annotationView = annotationContext.annotationView;
             if (annotationView && annotationView.enabled) {
-            {
-                // Annotations represented by views use the view frame as the positioning rect.
-                positioningRect = annotationView.frame;
-                [annotationView.superview bringSubviewToFront:annotationView];
-                [annotationView setSelected:YES animated:animated];
+                {
+                    // Annotations represented by views use the view frame as the positioning rect.
+                    positioningRect = annotationView.frame;
+                    [annotationView.superview bringSubviewToFront:annotationView];
+                    [annotationView setSelected:YES animated:animated];
+                }
             }
         }
-    }
 
-     // The client can request that any annotation be selected (even ones that are offscreen).
-     // The annotation can’t be selected if no part of it is hittable.
+    // The client can request that any annotation be selected (even ones that are offscreen).
+    // The annotation will bounce to the current tap point as anchor.
```

Comment:
This comment implies that the PR fixes #3249, but I don’t see anything that actually moves the camera. It’s the callout view that aligns with the tap point, not the annotation.

Reply:
What I'm trying to say is that if the annotation's center coordinate is outside the current viewport it will use the current tap position instead. Does this makes more sense?
```
// For annotations which `coordinate` falls offscreen it will use the current tap point as anchor instead.
```

Reply:
That depends on whether the annotation is a point annotation or shape annotation:

> The client can request that any annotation be selected, even an annotation that is off-screen. If a shape annotation is visible but its centroid is not, and a gesture recognizer is specified, the callout view is anchored to the gesture recognizer’s location.